libraries/Microsoft.Bot.Builder.Expressions/BuiltInFunctions.cs

```diff
@@ -1131,13 +1163,27 @@ private static DateTime ParseTimestamp(string timeStamp)
                 { ExpressionType.First, new ExpressionEvaluator(Apply(args =>
                     {
                         if (args[0] is string string0 && string0.Length > 0) return string0.First().ToString();
-                        if (args[0] is IList list && list.Count > 0) return list[0];
+                        if (args[0] is IList list && list.Count > 0)
```

Comment:
This is not the most efficient way to do JToken, if we do it this way, we will need to change everyfunction, you should call list.Element(index), and do the jvalue handle there

Reply:
to be more specific, we should not call list[0], we should call Element(list, 0)